src/Microsoft.OpenApi.Readers/OpenApiStreamReader.cs

```diff
@@ -15,7 +18,17 @@ namespace Microsoft.OpenApi.Readers
     /// </summary>
     public class OpenApiStreamReader : IOpenApiReader<Stream, OpenApiDiagnostic>
     {
+        private OpenApiReaderSettings _settings;
 
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="settings"></param>
+        public OpenApiStreamReader(OpenApiReaderSettings settings = null)
```

Comment:
it's better to separate two constructors?

Reply:
I don't think so.  The optional parameter conveys that settings are optional and having one constructor prevents someone from adding code to one ctor that needs to be called from both.

---
In reply to: [161353699](https://github.com/Microsoft/OpenAPI.NET/pull/176#discussion_r161353699) [](ancestors = 161353699)